src/enum.rs

```diff
@@ -72,6 +72,12 @@ impl Enum {
         self
     }
 
+    /// Add an arbitrary macro.
+    pub fn r#macro(&mut self, r#macro: &str) -> &mut Self {
+        self.type_def.r#macro(r#macro);
+        self
+    }
+
     /// Push a variant to the enum, returning a mutable reference to it.
```

Comment:
`<bike-shedding>`
Would it be better to have the function name not be a reserved keyword? it's a minor thing, but perhaps `annotation`?
`</bike-shedding>`